ClientApp/src/app/Services/user.service.ts

```diff
@@ -22,11 +24,27 @@ export class UserService {
   constructor(private http: HttpClient) {
   }
 
-  addUser(model: RegisterModel): Observable<Response> {
-    return this.http.post<Response>(this.usersUrl + '/register', model, this.httpOptions);
+  register(model: RegisterModel): Observable<ResponseModel> {
+    return this.http.post<ResponseModel>(this.usersUrl + '/register', model, this.httpOptions).pipe(
+      catchError(error => this.errorHandling(error))
+    );
   }
 
-  logIn(model): Observable<Response>{
-    return this.http.post<Response>(this.usersUrl + '/authenticate', model, this.httpOptions);
+  logIn(model): Observable<ResponseModel>{
+    return this.http.post<ResponseModel>(this.usersUrl + '/authenticate', model, this.httpOptions).pipe(
+      catchError(error => this.errorHandling(error))
+    );
+  }
+
+  get isLoggedIn(): boolean {
+    return localStorage.getItem('token') !== null;
+  }
+
+  logout(): void {
+    localStorage.removeItem('token');
+  }
+
+  errorHandling(error: HttpErrorResponse): Observable<any>{
```

Comment:
Name of any method should start with a verb. Something like 'handleError' will be better as a name.
Also missing space before {